src/app/services/views/ServiceListView.vue

```diff
@@ -176,7 +176,16 @@ async function loadService({ mesh, name }: { mesh: string, name: string }): Prom
   service.value = await kumaApi.getServiceInsight({ mesh, name })
 
   if (service.value.serviceType === 'external') {
-    externalService.value = await kumaApi.getExternalService({ mesh, name })
+    // The following code is a hotfix for https://github.com/kumahq/kuma-gui/issues/599 until we implement the lookup of `ExternalService` resources by `ServiceInsight` name.
+    const { items } = await kumaApi.getAllExternalServicesFromMesh({ mesh })
+
+    if (Array.isArray(items)) {
+      const foundExternalService = items.find((externalService) => externalService.tags['kuma.io/service'] === name)
+
+      if (foundExternalService !== undefined) {
+        externalService.value = foundExternalService
+      }
+    }
```

Comment:
This looks fine to me, but how about putting this code inside `kumaApi.getExternalService()` that way if it's needed anywhere else, the code is centralised in the API client. Plus, the fact that we need to do this at all is hidden by our api client. Lastly, this bug shows that the current implementation of `kumaApi.getExternalService()` doesn't work as intended.

Not a blocker at all

Reply:
I specifically don’t want to add it to the API wrapper because it insinuates that the API of retrieving an external service is broken which is not true. If you provide it with the name of an external service, it will work, but that’s not what we’ve been doing in the service views.

However, I noticed that we do actually need this in one other place (the service detail view) and so I externalized this code into a function which I intend to delete again once we fix this properly with the appropriate backend changes.